docs/DeveloperGuide.adoc

```diff
@@ -153,55 +153,157 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
-image::stClassDiagram.png[]
+[.text-justify]
+From Fig 1 below, each `Assignment` has four components: `Title`, `Deadline`, `Status` and `Workload`. `Workload` refers to the expected amount of time required to finish the `Assignment`. These components will be used for schedule generation when `(st)schedule n/NUM_DAYS` is executed (explained below).
 
-This feature enhances the basic form of the `SchoolworkTracker` by allowing a user to get a better sense of their current commitments and workload so that they can better manage their time to pursue other social events and hang out with friends. It utilises the existing `Assignments` stored in the `AssignmentList` of the `SchoolworkTracker`. Additionally, it implements the following operations:
+image::stClassDiagram.png[align="center"]
+[.text-center]
+_Fig 1. Class Diagram detailing the main components of the SchoolworkTracker_
 
-* `SchoolworkTracker#calExpectedWorkPerDay()` -- Calculates the expected amount of work the user has to put in each day for the next five days (based on hours) to meet all deadlines.
+=== Schedule
+==== Current Implementation
+image::scheduleResult.png[align="center"]
+[.text-center]
+_Fig 2. Sample result of the generated schedule_
 
-Given below is an example usage scenario of how the schedule visualizer behaves at each stage.
+[.text-justify]
+Schedule is facilitated by the `SchoolworkTracker`. This feature enhances the basic form of the `SchoolworkTracker` by allowing users to better visualise their current commitments so that they can better manage their time to pursue social events. This feature aims to distribute `Workload` as evenly as possible while ensuring that the user is able to complete the `Assignment` before the `Deadline`, unless it is impossible (i.e. `Workload` exceeds the amount of time the user has before the `Deadline` after taking into account more urgent assignments). A sample result is shown in Fig 2 above.
 
- Step 1. The user launches the application for the first time. The `SchoolworkTracker` will be initialized with the initial state, which is an empty list.
+[.text-justify]
+It utilizes the existing `Assignment` stored in the `SchoolworkTracker` and implements `ScheduleList` which represents the generated schedule.
 
-Step 2. The user executes the add assignment command [`(st)add t/CS2103 tP d/2020-03-27 12:00 e/5`] to add the following assignments to the Schoolwork Tracker.
+[.text-justify]
+As seen from Fig 3 below, `ScheduleList` consists of `Day` objects and each `Day` has 3 components. `totalAllocatedHours` represents the total number of hours allocated to the `Day`.
+`dueAssignment` refers to an assignment that is due on the `Day` and `allocatedAssignment` represents an assignment that has been scheduled to be done on that `Day`. The latter also has an additional field `allocatedHours` which correspond to the suggested amount of time
+that should be spent on the assignment on that `Day`.
```

Comment:
how come this is on a new line?

Reply:
Its not i think is just Github saying i added this line O: HAHA

Reply:
ohh wait erm when converted to pdf its not but i just separated it from the above line so its easier to read on intelliJ HAHA